http/middleware.go

```diff
@@ -1,14 +0,0 @@
-package http
-
-import (
-	"net/http"
-)
-
-type LoggingMiddleware struct {
-	Next http.Handler
-}
-
-func (l *LoggingMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	log.Infof("HTTP %s %v", r.Method, r.URL)
-	l.Next.ServeHTTP(w, r)
-}
```

Comment:
1) We're not using it in dex.
2) This implementation is tied to the capnslog package so it seemed reasonable to remove as part of this diff. It's basically a "log to capnslog middleware"